framework/src/meshmodifiers/ElementSubdomainModifierBase.C

```diff
@@ -0,0 +1,662 @@
+               "not cover the whole model, otherwise no elements will be reinitialized as it is "
+               "impossible for an element's old subdomain to not be in the list.");
+  else if (_old_subdomain_reinitialized == false && _subdomain_ids_to_reinitialize.empty())
+    paramError("old_subdomain_reinitialized",
+               "'old_subdomain_reinitialized' can only be set to false if "
+               "reinitialize_subdomains is set to a non-empty list of subdomains, otherwise no "
+               "elements will be reinitialized, as it is impossible for an element's new subdomain "
+               "to be in the list.");
+
+  auto bnd_names = getParam<std::vector<BoundaryName>>("moving_boundaries");
+  auto bnd_ids = _mesh.getBoundaryIDs(bnd_names, true);
+  const auto bnd_subdomains =
+      getParam<std::vector<std::vector<SubdomainName>>>("moving_boundary_subdomain_pairs");
+
+  if (bnd_names.size() == 1 && bnd_subdomains.size() > 1)
+  {
+    bnd_names.insert(bnd_names.end(), bnd_subdomains.size() - 1, bnd_names[0]);
+    bnd_ids.insert(bnd_ids.end(), bnd_subdomains.size() - 1, bnd_ids[0]);
+  }
+  else if (bnd_names.size() != bnd_subdomains.size())
+    paramError("moving_boundary_subdomain_pairs",
+               "Each moving boundary must correspond to a pair of subdomains. ",
+               bnd_names.size(),
+               " boundaries are specified by the parameter 'moving_boundaries', while ",
+               bnd_subdomains.size(),
+               " subdomain pairs are provided. Alternatively, if one boundary and multiple "
+               "subdomain pairs are provided, then the subdomain pairs all apply to one boundary.");
+
+  for (auto i : index_range(bnd_names))
+  {
+    _moving_boundary_names[bnd_ids[i]] = bnd_names[i];
+
+    if (bnd_subdomains[i].size() == 2)
+      _moving_boundaries[{_mesh.getSubdomainID(bnd_subdomains[i][0]),
+                          _mesh.getSubdomainID(bnd_subdomains[i][1])}] = bnd_ids[i];
+    else if (bnd_subdomains[i].size() == 1)
+      _moving_boundaries[{_mesh.getSubdomainID(bnd_subdomains[i][0]), Moose::INVALID_BLOCK_ID}] =
+          bnd_ids[i];
+    else
+      paramError("moving_boundary_subdomain_pairs",
+                 "Each subdomain pair must contain 1 or 2 subdomain names, but ",
+                 bnd_subdomains[i].size(),
+                 " are given.");
+  }
+}
+
+void
+ElementSubdomainModifierBase::modify(
+    const std::unordered_map<dof_id_type, std::pair<SubdomainID, SubdomainID>> & moved_elems)
+{
+  // Create moving boundaries on the undisplaced and displaced meshes
+  //
+  // Note: We do this _everytime_ because previous execution might have removed the sidesets and
+  // nodesets. Most of the moving boundary algorithms below assume that the moving sidesets and
+  // nodesets already exist on the mesh.
+  createMovingBoundaries(_mesh);
+  if (_displaced_mesh)
+    createMovingBoundaries(*_displaced_mesh);
+
+  // If nothing need to change, just return.
```

Comment:
can we move these 6 lines above? if we did not change the elements we didnt move the sideset/nodeset?

Reply:
Hello, I have swapped the order